A59-audit-logging.md

```diff
@@ -0,0 +1,349 @@
+the audit condition is met. This is analogus to the semantics that all access
+loggers will be run after the response is sent. It's a repeated field here since
+allowing multiple audit loggers is useful when users plan to roll out a new
+logger. By also keeping the existing logger along for a period they will be able
+to prevent interruption.
+
+Some readers may immediately notice that since an HTTP filter chain can contain
+an arbitrary number of RBAC filters, more than one of them could meet its audit
+condition for the same RPC and multiple audit log entries occur. When this happens,
+it would be impossible to uniquely identify a specific RPC from a log entry so as
+to apply any sort of de-duplication. The design in this gRFC does not intend to
+address this issue, the rationale of which will be elaborated more in
+[one of the design alternatives](#audit-condition-in-the-http-connection-manager).
+
+### gRPC Authorization Policy Changes
+
+The gRPC authorization policy consists of `deny_rules` and `allow_rules`, which
+are mapped to two RBACs. Overall it's a subset of the RBAC as stated in [A43: gRPC authorization API][A43].
+We will keep this philosophy by adding just one `audit_logging_options` in the
+policy instead of having separate ones for those two sets of rules.
+
+Following are changes to the JSON schema of gRPC Authorization Policy introduced
+in [A43: gRPC authorization API][A43]:
+
+```json
+{
+  "title": "AuthorizationPolicy",
+  "definitions": {
+    ...
+    "audit_logger": {
+      "description": "Configuration for the audit logger.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The name of the audit logger configuration."
+            "It is mainly used for human purposes.",
+          "type": "string",
+        },
+        "typed_config": {
+          "description": "The typed config for the audit logger."
+            "This needs to be a json object mapped from google.protobuf.Struct"
+            "proto message.",
+          "type": "object",
+        }
+      },
+    }
+  },
+  "properties": {
+    ...
+    "audit_logging_options": {
+      "description": "Audit logging options that include the condition for audit"
+        "logging to happen and audit logger configurations.",
+      "type": "object",
+      "properties": {
+        "audit_condition": {
+          "description": "Condition for the audit logging to happen."
+            "Same as NONE if ommited.",
+          "type": "string",
+          "enum": ["NONE", "ON_DENY", "ON_ALLOW", "ON_DENY_AND_ALLOW"]
+        },
+        "audit_logger": {
+          "description": "Audit logger configurations",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/audit_logger"
+          }
+        }
+      }
+    },
+  },
+  "required": ["name", "allow_rules"]
+}
+```
+
+For readers that are more familiar with proto messages, the change translates
+to:
+
+```
```

Comment:
FYI: You can use the \`\`\`proto (or "protobuf") here to get syntax highlighting. Ditto above.

Reply:
Done. Thank you, Eric!